src/librustc_incremental/persist/load.rs

```diff
@@ -169,6 +127,50 @@ pub fn load_dep_graph(sess: &Session) ->
     let report_incremental_info = sess.opts.debugging_opts.incremental_info;
     let expected_hash = sess.opts.dep_tracking_hash();
 
+    let mut prev_work_products = FxHashMap();
+
+    // If we are only building with -Zquery-dep-graph but without an actual
+    // incr. comp. session directory, we exit here. Otherwise we'd fail
+    // when trying to load work products.
```

Comment:
I'd reword this comment since we don't really exit the function if there is no cache directory. It could just say something like "If there is an incr. comp. cache directory, load the set of previous work products".